src/t8_schemes/t8_default/t8_default_common/t8_default_common.hxx

```diff
@@ -24,37 +24,179 @@
+
   /** Destructor for all default schemes */
-  virtual ~t8_default_scheme_common_c ();
+  ~t8_default_scheme_common ()
+  {
+    T8_ASSERT (ts_context != NULL);
+    SC_ASSERT (((sc_mempool_t *) ts_context)->elem_count == 0);
+    sc_mempool_destroy ((sc_mempool_t *) ts_context);
+  }
+
+  /** Move constructor */
+  t8_default_scheme_common (t8_default_scheme_common &&other) noexcept
+    : element_size (other.element_size), ts_context (other.ts_context), eclass (other.eclass)
+  {
+    other.ts_context = nullptr;
+  }
+
+  /** Move assignment operator */
+  t8_default_scheme_common &
+  operator= (t8_default_scheme_common &&other) noexcept
+  {
+    if (this != &other) {
+      // Free existing resources of moved-to object
+      if (ts_context) {
+        sc_mempool_destroy ((sc_mempool_t *) ts_context);
+      }
 
-  virtual void
-  t8_element_deinit (int length, t8_element_t *elem) const override;
+      // Transfer ownership of resources
+      element_size = other.element_size;
+      eclass = other.eclass;
+      ts_context = other.ts_context;
+
+      // Leave the source object in a valid state
+      other.ts_context = nullptr;
+    }
+    return *this;
+  }
+
+  /** Copy constructor */
+  t8_default_scheme_common (const t8_default_scheme_common &other)
+    : element_size (other.element_size), ts_context (sc_mempool_new (other.element_size)), eclass (other.eclass) {};
+
+  /** Copy assignment operator */
+  t8_default_scheme_common &
+  operator= (const t8_default_scheme_common &other)
+  {
+    if (this != &other) {
+      // Free existing resources of assigned-to object
+      if (ts_context) {
+        sc_mempool_destroy ((sc_mempool_t *) ts_context);
+      }
+
+      // Copy the values from the source object
+      element_size = other.element_size;
+      eclass = other.eclass;
+      ts_context = sc_mempool_new (other.element_size);
+    }
+    return *this;
+  }
 
   /** Compute the number of corners of a given element. */
   int
-  t8_element_num_corners (const t8_element_t *elem) const override;
+  element_get_num_corners (const t8_element_t *elem) const
+  {
+    /* use the lookup table of the eclasses.
+     * Pyramids should implement their own version of this function. */
+    return t8_eclass_num_vertices[eclass];
+  }
 
   /** Allocate space for a bunch of elements. */
   void
-  t8_element_new (int length, t8_element_t **elem) const override;
+  element_new (int length, t8_element_t **elem) const
+  {
+    t8_default_mempool_alloc ((sc_mempool_t *) ts_context, length, elem);
+  }
 
   /** Deallocate space for a bunch of elements. */
   void
-  t8_element_destroy (int length, t8_element_t **elem) const override;
+  element_destroy (int length, t8_element_t **elem) const
+  {
+    t8_default_mempool_free ((sc_mempool_t *) ts_context, length, elem);
+  }
+
+  void
+  element_deinit (int length, t8_element_t *elem) const
+  {
+  }
 
   /** Return the shape of an element */
   t8_element_shape_t
```

Comment:
Please add a comment that pyramids have their own implementation of this function.